File: MobileApp/my_social_app/lib/models/message_image.dart
import 'package:flutter/material.dart';
import 'package:json_annotation/json_annotation.dart';
import 'package:my_social_app/state/image_status.dart';
import 'package:my_social_app/state/message_entity_state/message_image_state.dart';
part 'message_image.g.dart';

@immutable
@JsonSerializable()
class MessageImage{
  final int id;
  final int messageId;
  final String blobName;
  final double height;
  final double width;

  const MessageImage({
    required this.id,
    required this.messageId,
    required this.blobName,
    required this.height,
    required this.width
  });

  factory MessageImage.fromJson(Map<String, dynamic> json) => _$MessageImageFromJson(json);
  Map<String, dynamic> toJson() => _$MessageImageToJson(this);

  MessageImageState toMessageImageState()
    => MessageImageState(
        id: id,
        messageId: messageId,
        blobName: blobName,
        height: height,
        width: width,
        image: null,
        status: ImageStatus.notStarted,
      );
}

File: src/MySocailApp.Domain/MessageAggregate/Entities/MessageImage.cs
using MySocailApp.Core;

namespace MySocailApp.Domain.MessageAggregate.Entities
{
    public class MessageImage(string blobName, double height, double width) : Entity
    {
        public int MessageId { get; private set; }
        public string BlobName { get; private set; } = blobName;
        public double Height { get; private set; } = height;
        public double Width { get; private set; } = width;
    }
}
